api-catalog-services/src/main/java/org/zowe/apiml/apicatalog/services/status/APIDocRetrievalService.java

```diff
@@ -314,9 +310,13 @@ private ApiInfo findApi(List<ApiInfo> apiInfos, String apiVersion) {
             return null;
         }
 
+        String api[] = apiVersion.split(" ");
+        String apiId = api[0];
+        String version = api[1].replace("v", "");
+
         return apiInfos.stream()
             .filter(
-                f -> f.getGatewayUrl().equals(apiVersion == null ? "api" : "api/" + apiVersion)
+                f -> f.getApiId().equals(apiId) && f.getVersion().equals(version)
             )
```

Comment:
in this place, it is more probable that apiId and version won't be null, so I would say it would be better to switch the equals